packages/runtime/src/useCases/consumption/openid4vc/RequestCredentials.ts

```diff
@@ -33,7 +35,20 @@ export class RequestCredentialsUseCase extends UseCase<RequestCredentialsRequest
     }
 
     protected override async executeInternal(request: RequestCredentialsRequest): Promise<Result<RequestCredentialsResponse>> {
-        const credentialResponses = await this.openId4VcController.requestCredentials(request.credentialOffer, request.credentialConfigurationIds, request.pinCode);
+        let token: OpenId4VciRequestTokenResponse | undefined;
+        if (request.accessToken !== undefined) {
+            // credo provides its own machanism to handle authentication and therefore expects the token to be in this format
+            // as we however can not use credo's solution from within the app calling this code, we need to manually create this structure here
```

Comment:
I don't see a reason for explaining this casting - additionally I'd like to have this in the controller method instead of the usecase.